compiler/jsgen.nim

```diff
@@ -179,9 +179,24 @@ proc mapType(p: PProc; typ: PType): TJSTypeKind =
   else: result = mapType(typ)
 
 proc mangleName(s: PSym; target: TTarget): Rope =
+  proc validJsName(name: string): bool =
+    result = true
+    const reservedWords = ["break", "case", "catch", "class", "const", "continue",
```

Comment:
The problem is, in nim you could still do:
```nim
type SomeType = object
  `case`: int
  `for`: cstring
  # ...
```
So unless one would make those names invalid field names for nim, they may not be removed from the list, I think.